test/test20.c

```diff
@@ -0,0 +1,61 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "../Annotations/Annotations.h"
+
+struct my_struct {
+    char* x; 
+    char* y;
+};
+
+void does_free(char* s) { 
+    free(s); 
+}
+
+
+char* creates_obligation(char* s, struct my_struct X) {    
+    free(s);  
+    free(X.x); 
+    char* str = (char*)malloc(15); 
+    return str; 
+}
+
+
+struct my_struct does_something(struct my_struct S) {
+    free(S.x); 
+    return S; 
+} 
+
+void does_malloc_calls(struct my_struct S) {
+    S.x = (char*)malloc(15);
+    S.y = (char*)malloc(15); 
+}
+
+void does_free_calls(struct my_struct S) {
+    free(S.x);
+    free(S.y); 
+}
+
+
+
+
+/*
+Right now I'd like the AT_PARAMETER and AT_FIELD to be 0-indexed 
+(I used 1-indexing before since I saw RLC use it with the field=#..., but just to be more consistent with LLVM I'll use 0-indexing) since if we had an annotation like AT_FIELD("x"), 
```

Comment:
Using 0-indexing is good. The RLC is 1-indexed to be consistent with the rest of the Checker Framework (the Java analysis tool that it is built on), which uses 1-indexing for parameters. (The reason that the CF uses 1-indexing for parameters is that most Java methods are instance methods of some object, so they implicitly have a "receiver" parameter---the object on which the method is called---which internally is stored as parameter 0. So, the RLC is sort of 0-indexed, too, but that's not visible externally.)